hooks/useVotingGuide.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  Exact,
+  useElectionVotingGuideByUserIdQuery,
+  useUpsertVotingGuideMutation,
+  VotingGuideResult,
+} from "generated";
+import { createContext, ReactNode, useContext } from "react";
+import { useQueryClient } from "react-query";
+
+type VotingGuideQueryContext = {
+  data: VotingGuideResult;
+  queryKey: (string | Exact<{ electionId: string; userId: string }>)[];
+};
+
+const VotingGuideContext = createContext<VotingGuideQueryContext>(
+  {} as VotingGuideQueryContext
+);
+
+export function VotingGuideProvider({
+  electionId,
+  userId,
+  children,
+}: {
+  electionId: string;
+  userId: string;
+  children: ReactNode;
```

Comment:
We can wrap this in `React.PropsWithChildren<>` and avoid needing to specify `children` explicitly

Reply:
good call